addons/account/models/account_payment.py

```diff
@@ -309,6 +309,20 @@ def open_payment_matching_screen(self):
             'context': action_context,
         }
 
+    @api.onchange('amount')
+    def _onchange_amount(self):
+        res = {}
+        journal_type = ['bank', 'cash']
+        if self.amount == 0:
+            journal_type.append('general')
+            self.payment_difference_handling = 'reconcile'
+            self.journal_id = self.env['account.journal'].search([('type', '=', 'bank')], limit=1)
```

Comment:
According to specification it is defined to select the first bank journal by default